src/test_util.rs

```diff
@@ -354,8 +357,21 @@ impl Context {
         let resource = Resource::from_both(&mock_tx, DummyLoader).map_err(Error::VerifyScript)?;
         let rtx = resolve_transaction(tx, &mut HashSet::new(), &resource, &resource)
             .map_err(|err| Error::VerifyScript(format!("Resolve transaction error: {:?}", err)))?;
-
-        let mut verifier = TransactionScriptsVerifier::new(&rtx, &resource);
+        let consensus = ConsensusBuilder::default()
+            .hardfork_switch(HardForks {
+                ckb2021: CKB2021::new_dev_default(),
+                ckb2023: CKB2023::new_dev_default(),
+            })
+            .build();
+        let tip = HeaderBuilder::default().number(0.pack()).build();
+        let tx_verify_env = TxVerifyEnv::new_submit(&tip);
```

Comment:
Is this the recommended way to build a `TxVerifyEnv` to verify txs offchain?

Reply:
No, this is only a test case `tip` should be the latest tip header, and `TxVerifyEnv` have `new_submit`,`new_commit` methods to get `Self`.

Reply:
You can refer to the implementation of the CKB's `estimate_cycles` RPC.

https://github.com/nervosnetwork/ckb/blob/e5b74d01b654a1a4b6cd74d28e19b0b92d630e0e/rpc/src/module/chain.rs#L2354

Reply:
OK. Thanks.